### ISSUE TITLE
Use packagename.classname as context name in text extractor

### ISSUE THREAD

**Issue by lassoan** [Slicer/sliceri18n#4]
Get class name where the `_()` function is located and the package name (name of the folder where the `__init__.py` file is located) and the context name is `<packagename>`.`<classname>`.

**Comment by syrjay** [comment]
hello @lassoan , so this __init__.py is the the file where I define the extract function ?
I thought that the packagename is the folder where located the .py file where we should extract strings...

**Comment by lassoan** [comment]
Yes, you are right, there are two different cases. Things can be confusing, because "Slicer module" and "Python module" are both called "module" but they mean slightly different things. To make it clear, hereby I'll refer to "Slicer module" = `smodule`; and "Python module" = `pmodule`.

1. Smodule

Can be implemented in C++ or Python. Slicer discovers them at startup and loads them from the list of folders designated as smodule folders. You can add additional smodule paths in application settings. If smodule is implemented Python then it is implemented in a single .py file, the **name of the smodule is always the same as the .py filename**.

**The context of a class in a smodule should be `<smodule name>.<class name>`.**

2. Pmodule

If a smodule becomes too complicated then it makes sense to split it up into multiple files. However, these files cannot just placed into the smodule folder, because Slicer would try to load each as a smodule. Instead, you create a pmodule inside the smodule's folder. The pmodule has a `__init__.py` file in its folder and can be used in the smodule by importing it (using `import pmodulename`). **The pmodule name is the name of the folder where the `__init__.py` is located.**

**The context of a class in a pmodule should be `<pmodule name>.<class name>`.**

I'm not sure what we should do when there is a class embedded in a class, maybe the context name should be then `<module name>.<class name>.<embedded class name>`. We can do whatever is the simplest. It may be OK if at the beginning we don't support i18n in classes that are embedded in other classes.

**Comment by syrjay** [comment]
right, I understand the difference, for exemple in the **DICOMBrowser.py** file located at **Slicer/Modules/Scripted/DICOMLib/DICOMBrowser.py** the **DICOMLib** folder contain an __init__.py file an also the DICOMBrowser.py class name is SlicerDICOMBrowser, different from the DataProbe.py file where the class name is the same name as DataProbe.py (smodule).
Finally I noticed that whether it is pmodule or smodule, we can just take the direct parent name of .py file as package_name,  and the class name will be different according to the type of module.
If it is not always the case, you will tell me.

**Comment by lassoan** [comment]
When Slicer is installed all smodules are copied into a single folder, so for smodules the module name cannot be the folder name but it must be the .py filename.
